### PR TITLE
Issue #3348272 - Fix config override cache gets for Social Comment Upload

### PR DESCRIPTION
**Pull request by bramtenhove**
## Problem
We're doing expensive config gets in overrides and this is also for Social Comment Upload.

## Solution
Move the expensive calls within the `if` statement to avoid it being executed more often than necessary.

## Issue tracker
https://www.drupal.org/project/social/issues/3348272

## Theme issue tracker
N/a

## How to test
- [ ] Set a breakpoint, check that it's executed quite often.
- [ ] Go to this branch, it's only executed a handful of times.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/a

## Release notes
We improved performance by reducing the number of config get calls in Social Comment Upload.

## Change Record
N/a

## Translations
N/a

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

File: modules/social_features/social_comment_upload/src/SocialCommentUploadConfigOverride.php [hunk 1]
```diff
@@ -41,15 +41,11 @@ public function __construct(ConfigFactoryInterface $config_factory) {
...
     $overrides = [];
 
-    // We don't need to add any fields if uploads are disabled.
-    // We use getEditable and getOriginal to avoid override loops.
-    if ($this->configFactory->getEditable('social_comment_upload.settings')->getOriginal('allow_upload_comments', TRUE) == FALSE) {
-      return $overrides;
-    }
-
     // Add field_group and field_comment_files.
+    // We don't need to add any fields if uploads are disabled, so let's check.
```
Comment (tbsiqueira) [inline modules/social_features/social_comment_upload/src/SocialCommentUploadConfigOverride.php:45]: Bram, I think that this comment should be "We need to add" right? I see that the if logic changed from == FALSE to == TRUE

File: modules/social_features/social_comment_upload/src/SocialCommentUploadConfigOverride.php [hunk 2]
```diff
@@ -92,8 +88,10 @@ public function loadOverrides($names) {
     }
 
     // Add field_comment_files.
+    // We don't need to add any fields if uploads are disabled, so let's check.
```
Comment (tbsiqueira) [inline modules/social_features/social_comment_upload/src/SocialCommentUploadConfigOverride.php:91]: Same comment from below, this should be a "We need" right?

**Review by tbsiqueira — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3343-zrffciimncfiqvmtsanqtpn77syv5ify.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6411e39fed44dc651029a5b7

**Pull request merged**